src/main/java/org/wso2/carbon/connector/util/AbstractAzureMediator.java

```diff
@@ -0,0 +1,232 @@
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.wso2.carbon.connector.util;
+
+import com.azure.storage.file.datalake.models.DataLakeRequestConditions;
+import com.azure.storage.file.datalake.models.LeaseAction;
+import org.apache.axis2.AxisFault;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseException;
+import org.apache.synapse.commons.json.JsonUtil;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.data.connector.ConnectorResponse;
+import org.apache.synapse.data.connector.DefaultConnectorResponse;
+import org.wso2.carbon.connector.core.AbstractConnector;
+
+import java.time.OffsetDateTime;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Abstract Azure Mediator class to handle common operations.
+ */
+public abstract class AbstractAzureMediator extends AbstractConnector {
+
+    @SuppressWarnings("unchecked")
+    public static <T> T parse(String value, Class<T> type) throws IllegalArgumentException {
+
+        if (type == Integer.class) {
+            return (T) Integer.valueOf(value);
+        } else if (type == Double.class) {
+            return (T) Double.valueOf(value);
+        } else if (type == Boolean.class) {
+            return (T) Boolean.valueOf(value);
+        } else if (type == String.class) {
+            return (T) value;
+        } else {
+            throw new IllegalArgumentException("Unsupported type: " + type);
+        }
+    }
+
+    /**
+     * Get the connection string for the storage account.
+     *
+     * @param accountName Storage account name.
+     * @param accountKey  Storage account key.
+     * @param protocol    Protocol.
+     * @return Connection string.
+     */
+    public static String getStorageConnectionString(String accountName, String accountKey, String protocol) {
+
+        return AzureConstants.PROTOCOL_KEY_PARAM + protocol + AzureConstants.SEMICOLON +
+                AzureConstants.ACCOUNT_NAME_PARAM + accountName + AzureConstants.SEMICOLON
+                + AzureConstants.ACCOUNT_KEY_PARAM + accountKey;
+    }
+
+    abstract public void execute(MessageContext messageContext, String responseVariable, Boolean overwriteBody);
+
+    @Override
+    public void connect(MessageContext messageContext) {
+
+        String responseVariable = getMediatorParameter(
+                messageContext, AzureConstants.RESPONSE_VARIABLE, String.class, false
+                                                      );
+        Boolean overwriteBody = getMediatorParameter(
+                messageContext, AzureConstants.OVERWRITE_BODY, Boolean.class, false);
+        execute(messageContext, responseVariable, overwriteBody);
+    }
+
+    protected <T> T getMediatorParameter(
+            MessageContext messageContext, String parameterName, Class<T> type, boolean isOptional) {
+
+        Object parameter = getParameter(messageContext, parameterName);
```

Comment:
Let's separate the common code in the getMediatorParameter and getProperty methods into a new method and use it there.